backend-rust/src/indexer.rs

```diff
@@ -973,13 +972,24 @@ VALUES
             .execute(tx.as_mut())
             .await?;
 
-        if let Some(prepared_event) = &self.prepared_event {
-            prepared_event.save(tx).await?;
-        }
+        self.prepared_event.save(tx).await?;
+
         Ok(())
     }
 }
 
+/// Different types of block item events that can be prepared and the
+/// status of the event (if the event was in a successful or rejected
+/// transaction).
+struct PreparedEventData {
+    /// The prepared event. Note: this is optional temporarily until we can
+    /// handle all events.
+    event:   Option<PreparedEvent>,
+    /// The status of the event (if the event belongs to a successful or
+    /// rejected transaction).
+    success: bool,
+}
```

Comment:
This data structure seems very strange to me.

Why not use `Result<PreparedEvent, E>` with some appropriate `E` instead of this separate `success` field? Can you have `event: Some(e)` at the same time as `success: false`? What about `event: None` and `success: true`? Maybe the `success` field should be a field on the prepared event instead of in this struct?

Reply:
The type was removed now, and we will introduce an enum down the line to represent events from failed transactions.